static/js/components/forms/SiteContentForm.tsx

```diff
@@ -76,6 +77,42 @@ export default function SiteContentForm({
     [setDirty]
   )
 
+  const scrollToError = (
+    errors: FormikErrors<SiteFormValues>,
+    formElement: HTMLElement
+  ) => {
+    const firstErrorField = configItem.fields.find(
+      field => errors[field.name] !== undefined
+    )
+    if (firstErrorField === undefined) {
+      throw Error("Could not find form field for error.")
+    }
+    const queries = [
+      `[name=${firstErrorField.name}]`,
+      /**
+       * react-select inputs don't have name property set, but they are siblings
+       * of a label with "for".
+       */
+      `label[for=${firstErrorField.name}] ~ * input`,
+      // fallback... we won't be able to focus it, but can at least scroll
+      `label[for=${firstErrorField.name}]`
+    ]
+    const errorElement = queries
+      .map(query => formElement.querySelector<HTMLElement>(query))
+      .find(el => el)
+    if (!errorElement) {
+      return
+    }
+    const { matches: prefersReducedMotion } = window.matchMedia(
+      "(prefers-reduced-motion: reduce)"
+    )
+    errorElement.scrollIntoView({
+      behavior: prefersReducedMotion ? "auto" : "smooth",
+      block:    "center"
+    })
+    errorElement.focus({ preventScroll: true })
+  }
```

Comment:
I think we could simplify this function a bit - instead of querying for the input element, I think we can get the same functionality by just looking for the first instance on the page of `.form-error` - the class that we put on our form error messages. If we did that I think we'd also remove any runtime dependencies for this function, so it could go into a library file and get some isolated tests

Reply:
yes that sounds much better 👍

Reply:
That is a ... much cleaner approach 😄